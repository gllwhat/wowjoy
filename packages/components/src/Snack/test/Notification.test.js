import React from 'react';
import renderer from 'react-test-renderer';
import Notification from '../Notification';

it('Notification render top 200', () => {
  expect(<Notification top={200} />).toMatchSnapshot();
});

it('Notification test', () => {
  const notification = renderer.create(<Notification top={200} />);
  const instance = notification.getInstance();
  instance.add({ key: 1, content: <div key={1}>1</div> });
  instance.add({ key: 2, content: <div key={2}>2</div> });
  expect(instance.state.notices.length).toBe(2);
  expect(instance.state.notices[0].content).toEqual(<div key={1}>1</div>);
  instance.remove(3);
  expect(instance.state.notices.length).toBe(2);
  instance.remove(1);
  expect(instance.state.notices.length).toBe(1);
  instance.destroy();
  expect(instance.state.notices.length).toBe(0);
});
