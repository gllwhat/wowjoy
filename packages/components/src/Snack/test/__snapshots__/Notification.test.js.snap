// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification render top 200 1`] = `
<Notification
  top={200}
/>
`;
